addons/mail/models/ir_mail_server.py

```diff
@@ -58,3 +63,16 @@ def _get_test_email_from(self):
         # no from_filter or from_filter is configured for a domain different that
         # the default_from of company's alias_domain -> fallback
         return super()._get_test_email_from()
+
+    @api.model
+    def _filter_mail_servers_fallback(self, servers):
+        return servers.filtered(lambda s: not s.owner_id)
+
+    def _check_forced_mail_server(self, mail_server, allow_archived, smtp_from):
+        super()._check_forced_mail_server(mail_server, allow_archived, smtp_from)
+
+        if mail_server.owner_id:
+            if email_normalize(smtp_from) != mail_server.from_filter:
```

Comment:
As we have the "owner_id", should we also check "email_normalize(smtp_from)" against "email_normalize(mail_server.owner_id.email)", to avoid using an outdated mail server?